packages/flutter_tools/lib/src/flutter_project_metadata.dart

```diff
@@ -235,7 +272,9 @@ class MigrateConfig {
     bool update = true,
     required Logger logger,
   }) {
-    final FlutterProject flutterProject = projectDirectory == null ? FlutterProject.current() : FlutterProject.fromDirectory(projectDirectory);
+    final FlutterProject flutterProject = projectDirectory == null ||
+        FlutterProject.current().directory.path == projectDirectory.path ?
```

Comment:
`FlutterProject.current()` is a global, which should be avoided in new code.

However, can the `projectDirectory` parameter be made required and non-nullable? it looks like at every call side we're providing one.